dbt/adapters/bigquery/column.py

```diff
@@ -82,7 +82,7 @@ def literal(self, value):
     def data_type(self) -> str:
         if self.dtype.upper() == "RECORD":
             subcols = [
-                "{} {}".format(col.name, col.data_type) for col in self.fields  # type: ignore[attr-defined]
+                "`{}` {}".format(col.name, col.data_type) for col in self.fields  # type: ignore[attr-defined]
```

Comment:
To avoid hardcoding backticks in too many places, I'd suggest using the [`quoted`](https://github.com/dbt-labs/dbt-bigquery/blob/cf60e3ea38ebf216e940944ec16627680adbf5c5/dbt/adapters/bigquery/column.py#L74-L76) property like [this](https://github.com/dbt-labs/dbt-bigquery/blob/cf60e3ea38ebf216e940944ec16627680adbf5c5/dbt/include/bigquery/macros/adapters.sql#L153):

```suggestion
                "{} {}".format(col.quoted, col.data_type) for col in self.fields  # type: ignore[attr-defined]
```